Userland/Libraries/LibJS/JIT/Compiler.h

```diff
@@ -23,104 +21,102 @@ class Compiler {
-        O(NewClass, new_class)                                                   \
-        O(CreateVariable, create_variable)                                       \
-        O(GetById, get_by_id)                                                    \
-        O(GetByValue, get_by_value)                                              \
-        O(GetGlobal, get_global)                                                 \
-        O(GetVariable, get_variable)                                             \
-        O(GetCalleeAndThisFromEnvironment, get_callee_and_this_from_environment) \
-        O(PutById, put_by_id)                                                    \
-        O(PutByValue, put_by_value)                                              \
-        O(Call, call)                                                            \
-        O(CallWithArgumentArray, call_with_argument_array)                       \
-        O(TypeofVariable, typeof_variable)                                       \
-        O(SetVariable, set_variable)                                             \
-        O(ContinuePendingUnwind, continue_pending_unwind)                        \
-        O(ConcatString, concat_string)                                           \
-        O(BlockDeclarationInstantiation, block_declaration_instantiation)        \
-        O(SuperCallWithArgumentArray, super_call_with_argument_array)            \
-        O(GetIterator, get_iterator)                                             \
-        O(IteratorNext, iterator_next)                                           \
-        O(IteratorResultDone, iterator_result_done)                              \
-        O(ThrowIfNotObject, throw_if_not_object)                                 \
-        O(ThrowIfNullish, throw_if_nullish)                                      \
-        O(IteratorResultValue, iterator_result_value)                            \
-        O(IteratorClose, iterator_close)                                         \
-        O(IteratorToArray, iterator_to_array)                                    \
-        O(Append, append)
-
-#    define DECLARE_COMPILE_OP(OpTitleCase, op_snake_case) \
-        void compile_##op_snake_case(Bytecode::Op::OpTitleCase const&);
+    Assembler::Reg GPR0 = Assembler::Reg::empty();
+    Assembler::Reg GPR1 = Assembler::Reg::empty();
+    Assembler::Reg ARG0 = Assembler::Reg::empty();
+    Assembler::Reg ARG1 = Assembler::Reg::empty();
+    Assembler::Reg ARG2 = Assembler::Reg::empty();
+    Assembler::Reg ARG3 = Assembler::Reg::empty();
+    Assembler::Reg ARG4 = Assembler::Reg::empty();
+    Assembler::Reg ARG5 = Assembler::Reg::empty();
+    Assembler::Reg RET = Assembler::Reg::empty();
+    Assembler::Reg STACK_POINTER = Assembler::Reg::empty();
+    Assembler::Reg REGISTER_ARRAY_BASE = Assembler::Reg::empty();
+    Assembler::Reg LOCALS_ARRAY_BASE = Assembler::Reg::empty();
+    Assembler::Reg UNWIND_CONTEXT_BASE = Assembler::Reg::empty();
```

Comment:
The move of these from compile-time constants to member variable accesses is very unfortunate. 

Can we do something like:

```
#if ARCH(X86_64)
#include <LibJIT/X86_64/Assembler.h>
using ArchAssembler = X86_64::Assembler;
#elif ARCH ....
#endif
```

And have these generic reg names defined on the arch-specific assembler that we can just delegate to? 
```
struct X86Thingy {
  static constexpr auto GPR0 = 4;
};

using ArchThingy = X86Thingy;


struct GenericThingy {
   static constexpr auto GPR0 = ArchThingy::GPR0; 
};
```